src/track/track.cpp

```diff
@@ -914,29 +921,33 @@ void Track::setCuePoints(const QList<CuePointer>& cuePoints) {
 }
 
 Track::ImportStatus Track::importBeats(
-        mixxx::BeatsImporterPointer pBeatsImporter) {
+        mixxx::BeatsImporterPointer pBeatsImporter,
+        bool bpmLocked) {
     QMutexLocker lock(&m_qMutex);
     VERIFY_OR_DEBUG_ASSERT(pBeatsImporter) {
         return ImportStatus::Complete;
     }
+    if (m_record.getBpmLocked()) {
+        return ImportStatus::Complete;
+    }
+
     DEBUG_ASSERT(!m_pBeatsImporterPending);
     m_pBeatsImporterPending = pBeatsImporter;
     if (m_pBeatsImporterPending->isEmpty()) {
-        // Just return the current import status without clearing any
-        // existing cue points.
         m_pBeatsImporterPending.reset();
         return ImportStatus::Complete;
     } else if (m_record.hasStreamInfoFromSource()) {
-        // Replace existing cue points with imported cue
-        // points immediately
+        // Replace existing beats with imported beats immediately
         importPendingBeatsMarkDirtyAndUnlock(&lock);
+        setBpmLocked(bpmLocked);
```

Comment:
Ideally setBpmLocked() would be called within the locking scope. Please add comments why this must be set **after** importPendingBeatsMarkDirtyAndUnlock
/setBeatsMarkDirtyAndUnlock. Otherwise someone might reorder the statements to avoid double locking.